operator/api/v1beta1/helmrelease_types.go

```diff
@@ -40,7 +54,11 @@ type HelmReleaseSpec struct {
 type HelmReleaseStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
-	Code release.Status_Code `json:"code"`
+
+	Code               release.Status_Code     `json:"code"`
+	LastTransitionTime metav1.Time             `json:"lastTransitionTime,omitempty"`
```

Comment:
Should we model this like other kubernetes resources and put this under a `conditions` field, like how `Deployment` does for example:

```
status:
  conditions:
  - lastTransitionTime: 2019-07-24T20:02:35Z
    lastUpdateTime: 2019-08-09T14:14:03Z
    message: ReplicaSet "track-service-bfd88bdd8" has successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  - lastTransitionTime: 2019-08-09T17:06:37Z
    lastUpdateTime: 2019-08-09T17:06:37Z
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available

Reply:
I like making the HelmRelease's status more similar to other k8s resources, especially since what I had already looks really similar to these conditions. But I'm wondering if multiple conditions makes sense for HelmReleases.

Right now our conditions are basically the release's status code, and those are all mutually exclusive. A release is either FAILED/DEPLOYED/DELETED etc, but would never be in multiple of these states at once. 

Unless we wanted to add the `status: "True"/"False"` fields like your snippet has, but they seem redundant to me. If a condition is true, it will be present in the status. If it's false, it will be removed from the status conditions. Is there a use case they support that I'm not thinking of?